liquidprompt

```diff
@@ -1137,6 +1140,27 @@ _lp_jobcount_color() {
     [[ -n "$lp_jobcount_color" ]]
 }
 
+###################
+# Directory stack #
+###################
+
+_lp_dirstack() {
+    (( LP_ENABLE_DIRSTACK )) || return 2
+
+    lp_dirstack=
+
+    local -i ds
+    (( ds = $(dirs -p | wc -l) ))
+
+    if (( ds > 1 )); then
+        lp_dirstack+="${LP_COLOR_DIRSTACK}${LP_MARK_DIRSTACK}${ds}${NO_COL}"
+    else
+        return 1
+    fi
+
+    return 0
+}
```

Comment:
The v2.0 method is to separate the default theme from the data source.
Don't use `wc -l`, its is slower. Use the new `__lp_line_count()` function.
If you declare a variable as int, you don't need to use a math context to assign to it.

Here is how I would do this:

```suggestion
_lp_dirstack() {
    (( LP_ENABLE_DIRSTACK )) || return 2
    
    local count dir_stack=$(dirs -p; printf x)
    __lp_line_count "${dis_stack%x}"
    lp_dirstack=$count
    
    (( lp_dirstack > 1 ))
}

_lp_dirstack_color() {
    _lp_dirstack || return "$?"
    
    lp_dirstack_color="${LP_COLOR_DIRSTACK}${LP_MARK_DIRSTACK}${lp_dirstack}${NO_COL}"
}
```